syncstorage/db.go

```diff
@@ -887,7 +887,14 @@ func (d *DB) getBSOs(
 	}
 
 	limitStmt := "LIMIT ?"
-	values = append(values, limit)
+
+	// fetch an extra row to detect if there are more
+	// rows that match the query conditions
+	if limit > 0 {
+		values = append(values, limit+1)
+	} else {
+		values = append(values, limit)
+	}
```

Comment:
Will this do the right thin in the case of `limit=0` on a collection with no items?  Or, should the conditional here be `limit >=0`?